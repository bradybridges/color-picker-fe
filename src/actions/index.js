export const addProjects = (projects) => ({
  type: 'ADD_PROJECTS',
  projects,
});

export const addPalettes = (palettes) => ({
  type: 'ADD_PALETTES',
  palettes,
});
