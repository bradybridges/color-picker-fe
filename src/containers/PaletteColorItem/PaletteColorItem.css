.color-container {
  margin: 15px;
  text-align: center;
  width: 8%;
  height: 90%; 
}

.color-code {
  font-size: 1.5em;
}

.color-lock-img {
  width: 25%;
  /* position: relative; */
}