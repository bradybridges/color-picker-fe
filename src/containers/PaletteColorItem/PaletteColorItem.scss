.color-container {
  border-radius: 50%;
  border: solid #fff 1px;
  margin: 15px;
  text-align: center;
  min-width: 200px;
  max-width: 200px;
  height: 200px;
  &:hover {
    cursor: pointer;
  }
}

.color-code {
  font-size: 1.5em;
  margin: 65px 20px 0px 20px;
}

img {
  width: 25%;
  margin: 10px auto;
  /* position: relative; */
}