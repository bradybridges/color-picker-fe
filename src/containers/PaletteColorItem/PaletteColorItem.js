import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setTempPalette } from '../../actions/index';
import './PaletteColorItem.css'

const lockedImg = require('../../images/lock.svg');
const unlockedImg = require('../../images/unlocked-padlock.svg');

export class PaletteColorItem extends Component {
  handleToggleColorLock = () => {
    const { color, isLocked } = this.props;
    const updatedPalette = this.props.tempPalette.map((color) => color);
    const colorToUpdate = updatedPalette.find((palItem) => {
      if(palItem.color === color && palItem.isLocked === isLocked) {
        return true;
      } else { return false; }
    });
    colorToUpdate.isLocked = !colorToUpdate.isLocked;
    this.props.setTempPalette(updatedPalette);
  };

  render() {
    const { color, isLocked } = this.props;
    const lockImg = isLocked ? lockedImg : unlockedImg;
    const style = { backgroundColor: color }
    return (
      <section style={style} className='color-container'>
        <div className='color-container-contents'>
          <p className='color-code'>{color}</p>
          <img src={lockImg} className='color-lock-img' alt="lock or unlock color" onClick={this.handleToggleColorLock}/>
        </div>
      </section>
    );
  }
}

export const mapState = (state) => ({
  tempPalette: state.tempPalette,
});

export const mapDispatch = (dispatch) => ({
  setTempPalette: (palette) => dispatch(setTempPalette(palette)),
});

export default connect(mapState, mapDispatch)(PaletteColorItem);
