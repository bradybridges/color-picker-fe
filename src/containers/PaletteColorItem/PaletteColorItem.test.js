import React from 'react';
import { shallow } from 'enzyme';
import { setTempPalette } from '../../actions/index';
import { PaletteColorItem, mapState, mapDispatch } from './PaletteColorItem';

describe('PaletteColorItem', () => {
  let wrapper;
  const mockTemp = [
    {
      name: 'color_1',
      color: '#FFFFFF',
      isLocked: false,
    },
    {
      name: 'color_2',
      color: '#000000',
      isLocked: false,
    },
    {
      name: 'color_3',
      color: '#CCCCCC',
      isLocked: true,
    },
    {
      name: 'color_4',
      color: '#1F1F1F',
      isLocked: true,
    },
    {
      name: 'color_5',
      color: '#1D1D1D',
      isLocked: true,
    },
  ];
  const mockSetTempPalette = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<PaletteColorItem
      color='#CCCCCC'
      isLocked={true}
      tempPalette={mockTemp}
      setTempPalette={mockSetTempPalette}
    />);
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('handleToggleColorLock should toggle the state of isLocked', () => {
    wrapper.instance().handleToggleColorLock();
    const expected = [
      { color: '#FFFFFF', isLocked: false, name: 'color_1' },
      { color: '#000000', isLocked: false, name: 'color_2' },
      { color: '#CCCCCC', isLocked: false, name: 'color_3' },
      { color: '#1F1F1F', isLocked: true, name: 'color_4' },
      { color: '#1D1D1D', isLocked: true, name: 'color_5' },
    ];
    expect(mockSetTempPalette).toHaveBeenCalledWith(expected);
  });
});

describe('mapState', () => {
  it('should map tempPalette to props', () => {
    const mockTemp = [
      {
        name: 'color_1',
        color: '#FFFFFF',
        isLocked: false,
      },
      {
        name: 'color_2',
        color: '#000000',
        isLocked: false,
      },
      {
        name: 'color_3',
        color: '#CCCCCC',
        isLocked: false,
      },
      {
        name: 'color_4',
        color: '#1F1F1F',
        isLocked: true,
      },
      {
        name: 'color_5',
        color: '#1D1D1D',
        isLocked: true,
      },
    ];
    const mockState = {
      error: 'Something terrible happened',
      tempPalette: mockTemp,
    };
    const expected = { tempPalette: mockTemp };
    const mappedState = mapState(mockState);
    expect(mappedState).toEqual(expected);
  });
});

describe('mapDispatch', () => {
  it('should map setTempPalette to props and dispatch the correct action', () => {
    const mockDispatch = jest.fn();
    const mockTemp = [
      {
        name: 'color_1',
        color: '#FFFFFF',
        isLocked: false,
      },
      {
        name: 'color_2',
        color: '#000000',
        isLocked: false,
      },
      {
        name: 'color_3',
        color: '#CCCCCC',
        isLocked: false,
      },
      {
        name: 'color_4',
        color: '#1F1F1F',
        isLocked: true,
      },
      {
        name: 'color_5',
        color: '#1D1D1D',
        isLocked: true,
      },
    ];
    const actionToDispatch = setTempPalette(mockTemp);
    const mappedDispatch = mapDispatch(mockDispatch);
    mappedDispatch.setTempPalette(mockTemp);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
