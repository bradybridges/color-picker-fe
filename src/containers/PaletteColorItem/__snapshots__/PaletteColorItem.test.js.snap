// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteColorItem should match snapshot 1`] = `
<section
  className="color-container"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#CCCCCC",
    }
  }
>
  <div
    className="color-container-contents"
  >
    <p
      className="color-code"
    >
      #CCCCCC
    </p>
    <img
      alt="lock or unlock color"
      className="color-lock-img"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "lock.svg",
        }
      }
    />
  </div>
</section>
`;
